### ISSUE TITLE
Summary for IAM documentation

### ISSUE THREAD

**Issue by kaihendry** [apex/up#277]
Would be nice to flesh out https://github.com/apex/up/blob/master/docs/aws-credentials.md#iam-policy-for-up-cli maybe with a table like https://s.natalian.org/2017-08-28/travis-iam.png

Basically there are "adoption questions" surrounding these **super user** third party policies in many companies, and I'm just trying to share my experiences here, so they can be better answered.

I posted over at Serverless with similar concerns:
https://forum.serverless.com/t/policy-for-travis-ci-deployments/2021

Not sure if #82 might answer the **who** question. Basically the way I see it, is that you end up having to trust the third party tool aka `apex up` to ensure resources don't get abused. Seems a bit abstract, but I guess there is no way for AWS IAM to be the only gateway here IIUC, since it's painful to toggle/approve permissions whilst the third party tool is working?

**Comment by tj** [comment]
Yeah definitely, for now I have `*` just so it's less annoying for people to set up and re-configure often, but long-term having Up create some IAM roles for things like "up developer" (access to deploys only?), "up admin" (access to everything necessary to alter stacks) might be nice.

That said I imagine bigger companies will just use Up in a CI.

**Comment by kaihendry** [comment]
https://github.com/serverless/serverless/issues/3084#issuecomment-324679055 nicely summarises it as {deployment, runtime} roles.

Some super pedantic businesses might want to sign off on the deployment roles. Though I :heart: how simple https://github.com/apex/up/blob/master/docs/02-aws-credentials.md#iam-policy-for-up-cli is right now. It's a [complete mess](https://serverless.com/framework/docs/providers/aws/guide/iam/) over at [serverless framework](https://github.com/serverless/serverless/issues/3084#issuecomment-324628566) imo for instrumenting Travis CI deployments!

**Comment by tj** [comment]
ahah jesus yea that looks a little intense. I like the idea of making the IAM stuff more declarative at some point, possibly having Up create a few more roles, but it's a little early
